tests/s3_data_plane_tests.c

```diff
@@ -2675,50 +2675,57 @@ static int s_test_s3_round_trip_default_get_fc(struct aws_allocator *allocator,
     struct aws_byte_buf path_buf;
     AWS_ZERO_STRUCT(path_buf);
 
-    ASSERT_SUCCESS(aws_s3_tester_upload_file_path_init(
-        allocator, &path_buf, aws_byte_cursor_from_c_str("/prefix/round_trip/test_default_fc.txt")));
+    /* I know it's ugly, but I don't want to waste time to refact all those flexible checksum tests */
```

Comment:
~to refact~ -> refactoring? to refactor?

but more importantly, I don't know what "it" is, why it's ugly, or what "all those" other tests are. please make this comment make sense to a newcomer

are you talking about the loop is ugly? a loop seems ok to me

but if you wanted N explicit tests, instead of 1 test that checks all algorithms, just make this a helper function, and set up N tests that call it

Reply:
I would image we can have something like a helper to handle different type and different checksum algorithm instead of lot of really similar code around those flexible checksum tests and for loops with lot of code.